app/controllers/application_controller.rb

```diff
@@ -50,8 +50,18 @@ def decorated_current_user
     @decorated_current_user ||= current_user.decorate
   end
 
+  def decorated_true_user
+    @decorated_true_user ||= true_user.decorate
+  end
+
   def current_user
-    @current_user ||= User.find(session[:user_id]) if session[:user_id]
+    return @current_user if defined?(@current_user)
+
+    @current_user = if true_user.try(:staff?) && session[:impersonated_user_id]
```

Comment:
Is there ever a case where `true_user` would be `nil`? Can we avoid the call to `try`?